openmoc/process.py

```diff
@@ -918,6 +960,65 @@ def tally_fission_rates(self, solver, volume='integrated'):
 
         return tally
 
+
+    def tally_reaction_rates_on_mesh(self, solver, rxn_type, domain_type,
+                                     volume='integrated', energy='integrated'):
+        """Compute 'material' or 'cell' reaction rates on a mesh
+        
+        This method streamlines the process of tallying reaction rates on a
+        mesh by constructing the `domains_to_coeffs' dictionary and wrapping
+        the Mesh.tally_on_mesh() method.
+        
+        NOTE: This method assumes that the mesh perfectly aligns with the
+        flat source region mesh used in the OpenMOC calculation.
+        
+        Parameters
+        ----------
+        solver : {openmoc.CPUSolver, openmoc.GPUSolver, openmoc.VectorizedSolver}
+            The solver used to compute the flux
+        rxn_type : {'flux', 'total', 'scatter', 'fission', 'nu-fission'}
+            The cross section or reaction type to tally
+        domain_type : {'cell', 'material'}
+            The type of domain for which the coefficients are defined
+        volume : {'averaged', 'integrated'}
+            Compute volume-averaged or volume-integrated tallies
+        energy : {'by_group', 'integrated'}
+            Compute tallies by energy group or integrate across groups
+
+        Returns
+        -------
+        tally : numpy.ndarray of Real
+            A NumPy array of the fission rates tallied in each mesh cell indexed
+            by FSR ID and energy group (if energy is 'by_group')
+        
+        """
+        global solver_types, rxn_types
+        cv.check_type('solver', solver, solver_types)
+        cv.check_value('rxn_type', rxn_type, rxn_types)
+        cv.check_value('domain_type', domain_type, ('cell', 'material'))
+        cv.check_value('volume', volume, ('averaged', 'integrated'))
```

Comment:
Keep only the type checking that you need for this function. It's done again in tally_on_mesh anyway

Reply:
fixed